test/app-tap/datetime.test.lua

```diff
@@ -375,6 +379,95 @@ test:test("Formatting limits", function(test)
     end
 end)
 
+test:test("Simple tests for parser", function(test)
+    test:plan(4)
+    test:ok(date.parse("1970-01-01T01:00:00Z") ==
+            date.new{year=1970, mon=1, day=1, hour=1, min=0, sec=0})
+    test:ok(date.parse("1970-01-01T02:00:00+02:00") ==
+            date.new{year=1970, mon=1, day=1, hour=2, min=0, sec=0, tzoffset=120})
+
+    test:ok(date.parse("1970-01-01T02:00:00Z") <
+            date.new{year=1970, mon=1, day=1, hour=2, min=0, sec=1})
+    test:ok(date.parse("1970-01-01T02:00:00Z") <=
+            date.new{year=1970, mon=1, day=1, hour=2, min=0, sec=0})
+end)
+
+test:test("Multiple tests for parser (with nanoseconds)", function(test)
+    test:plan(193)
+    -- borrowed from p5-time-moments/t/180_from_string.t
```

Comment:
Perhaps it's worth to add a link to source of testcases to commit message (https://github.com/chansen/p5-time-moment).

Reply:
Yup. Sort of. Added comment to the test itself...

```diff
diff --git a/test/app-tap/datetime.test.lua b/test/app-tap/datetime.test.lua
index 007107509..803070358 100755
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -393,10 +393,18 @@ test:test("Simple tests for parser", function(test)
 end)
 
 test:test("Multiple tests for parser (with nanoseconds)", function(test)
-    test:plan(193)
-    -- borrowed from p5-time-moments/t/180_from_string.t
+    test:plan(211)
+    -- borrowed from
+    -- github.com/chansen/p5-time-moment/blob/master/t/180_from_string.t
     local tests =
     {
+        --{ iso-8601 string, epoch, nanoseconds, tz-offset, do reverse check?}
+        {'0001-01-01T00:00:00Z',    -62135596800,         0,    0, 0},
+        {'00010101T000000Z',        -62135596800,         0,    0, 0},
+        {'0001-W01-1T00:00:00Z',    -62135596800,         0,    0, 0},
+        {'0001W011T000000Z',        -62135596800,         0,    0, 0},
+        {'0001-001T00:00:00Z',      -62135596800,         0,    0, 0},
+        {'0001001T000000Z',         -62135596800,         0,    0, 0},
         {'1970-01-01T00:00:00Z',               0,         0,    0, 1},
         {'1970-01-01T02:00:00+0200',           0,         0,  120, 1},
         {'1970-01-01T01:30:00+0130',           0,         0,   90, 1},
```